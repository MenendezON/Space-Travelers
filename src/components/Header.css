header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 20px;
  background-color: #fff;
  border-bottom: 1px solid #ddd;
}

header > * {
  margin: 0 3%;
}

header h1 {
  padding: 20px 0 20px 100px;
  font-size: 30px;
  font-weight: 400;
  color: #000;
  background: url(planet1.png) top left/contain no-repeat;
}

.navbar ul {
  display: flex;
  list-style: none;
  text-transform: uppercase;
}

.navbar a {
  padding: 0 20px;
  color: #02bab7;
  text-decoration: none;
}

.navbar ul li:last-of-type {
  border-right: none;
}

.navbar a:hover,
.navbar .active {
  color: #9ad0cf;
  border-bottom: 2px solid #02bab7;
}

.profile-link {
  border-left: 1px solid #333;
}

@media (max-width: 960px) {
  header {
    flex-direction: column;
    padding: 20px 0;
  }

  header h1 {
    padding: 10px 0 10px 150px;
    font-size: 30px;
    font-weight: 400;
    color: #000;
    background: url(planet1.png) top left/contain no-repeat;
  }

  .navbar ul {
    padding: 0;
  }

  .navbar a {
    padding: 0 10px;
    font-size: 12px;
  }

  .profile-link {
    padding-top: 2px;
    border-left: 1px solid #333;
  }
}
