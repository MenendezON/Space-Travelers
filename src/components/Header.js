import './Header.css';
import { NavLink } from 'react-router-dom';

function Header() {
  return (
    <header>
      <h1>Space Travelers&apos; Hub</h1>
      <nav className="navbar">
        <ul>
          <li>
            <NavLink
              to="/"
              className={({ isActive }) => (isActive ? 'active link' : 'pending link')}
            >
              Rockets
            </NavLink>
          </li>
          <li>
            <NavLink
              to="missions"
              className="link"
            >
              Missions
            </NavLink>
          </li>
          <li>
            <NavLink
              to="dragons"
              className="link"
            >
              Dragons
            </NavLink>
          </li>
          <li>
            <NavLink
              to="profile"
              className="link profile-link"
            >
              My profile
            </NavLink>
          </li>

        </ul>
      </nav>
    </header>
  );
}
export default Header;
