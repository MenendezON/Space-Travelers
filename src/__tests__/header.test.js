import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from '../components/Header';
import '@testing-library/jest-dom/extend-expect';

describe('Header component', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Router>
        <Header />
      </Router>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should have active link for Rockets', () => {
    const { getByText } = render(
      <Router>
        <Header />
      </Router>,
    );

    const rocketsLink = getByText('Rockets');
    expect(rocketsLink).toBeInTheDocument();
  });
});
