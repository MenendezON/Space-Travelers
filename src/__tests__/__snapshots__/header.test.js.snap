// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render correctly 1`] = `
<div>
  <header>
    <h1>
      Space Travelers' Hub
    </h1>
    <nav
      class="navbar"
    >
      <ul>
        <li>
          <a
            aria-current="page"
            class="active link"
            href="/"
          >
            Rockets
          </a>
        </li>
        <li>
          <a
            class="link"
            href="/missions"
          >
            Missions
          </a>
        </li>
        <li>
          <a
            class="link"
            href="/dragons"
          >
            Dragons
          </a>
        </li>
        <li>
          <a
            class="link profile-link"
            href="/profile"
          >
            My profile
          </a>
        </li>
      </ul>
    </nav>
  </header>
</div>
`;
